src/realm/group_writer.cpp

```diff
@@ -462,10 +518,23 @@ ref_type GroupWriter::write_group()
     ref_type top_ref = free_versions_ref + free_versions_size;
 
     // Update top to point to the calculated positions
-    top.set(3, from_ref(free_positions_ref));                // Throws
-    top.set(4, from_ref(free_sizes_ref));                    // Throws
-    top.set(5, from_ref(free_versions_ref));                 // Throws
-    top.set(6, RefOrTagged::make_tagged(m_current_version)); // Throws
+    top.set(Group::s_free_pos_ndx, from_ref(free_positions_ref));               // Throws
+    top.set(Group::s_free_size_ndx, from_ref(free_sizes_ref));                  // Throws
+    top.set(Group::s_free_version_ndx, from_ref(free_versions_ref));            // Throws
+    top.set(Group::s_version_ndx, RefOrTagged::make_tagged(m_current_version)); // Throws
+    if (m_evacuation_limit == 0) {
+        size_t used_space = m_logical_size - m_free_space_size;
+        if (m_logical_size >= 0x100000 && m_free_space_size - m_locked_space_size > 2 * used_space) {
```

Comment:
I'd prefer magic constants to be replaced with named magic consts accompanied with explanatory comments about how they were chosen.

Reply:
Top